src/lib.rs

```diff
@@ -313,28 +317,28 @@ impl OctocrabBuilder {
     }
 
     /// Set the base url for `Octocrab`.
-    pub fn base_url(mut self, base_url: impl reqwest::IntoUrl) -> Result<Self> {
-        self.base_url = Some(base_url.into_url().context(crate::error::HttpSnafu)?);
```

Comment:
This should use `TryFrom<Url>` instead of `AsRef`, because then we can still accept `Url` in addition to `&str`.

Reply:
Looking at the docs: https://doc.rust-lang.org/std/convert/trait.TryInto.html
> Library authors should usually not directly implement this trait, but should prefer implementing the [TryFrom](https://doc.rust-lang.org/std/convert/trait.TryFrom.html) trait, which offers greater flexibility and provides an equivalent TryInto implementation for free, thanks to a blanket implementation in the standard library.

Reply:
>  but you can still use TryInto where appropriate

Can you tell the syntax to this please?

Reply:
It's the same as everything else.
```rust
fn foo(url: impl TryInto<Url>) -> Result<()> {
     let url = url.try_into()?;
     Ok(())
}
```

Reply:
I tried to use it, however I could not get the types to match up.